client/src/components/Account/Sidebar/Sidebar.tsx

```diff
@@ -0,0 +1,78 @@
+import { Box, List, ListItem, ListItemText, Typography } from '@material-ui/core';
+import { Link } from 'react-router-dom';
+import { useLocation } from 'react-router';
+
+import { useStyles } from './useStyles';
+
+export default function SideBar(): JSX.Element {
+  const classes = useStyles();
+  const { pathname } = useLocation();
+  const path = pathname.split('/')[2];
```

Comment:
instead of splitting the pathname like that because it's always tricky if you change your route pattern a little bit it would be hard to track this logic, i would recommend just sticking with the pathname so if it's /user/editprofile for example, i would compare the path comparison to path === "/user/editprofile" 

it makes it easier for maintainability :)

Reply:
Thank you, updated as per recommendation :)